### DISCUSSION TITLE
mapping entPhysicalIndex to IF Interface

### DISCUSSION THREAD

**Discussion by j4mbob** [prometheus/snmp_exporter#931] [Q&A]
Hi all,

wonder if anyone can help me.

I'm polling a huawei device that has a mib that deals with optical light levels  on sfp fibre interfaces.

the MIB uses 'entPhysicalIndex' from ENTITY-MIB as the index for each of the interfaces, which produces index's like these: (67108867)

```
.1.3.6.1.2.1.47.1.1.1.1.12.67108867 = STRING: Huawei
.1.3.6.1.2.1.47.1.1.1.1.12.67108869 = STRING: Huawei
.1.3.6.1.2.1.47.1.1.1.1.12.67108873 = STRING: Huawei
.1.3.6.1.2.1.47.1.1.1.1.12.67125260 = STRING: Huawei
.1.3.6.1.2.1.47.1.1.1.1.12.67141644 = STRING: Huawei
.1.3.6.1.2.1.47.1.1.1.1.12.67141645 = STRING: Huawei
.1.3.6.1.2.1.47.1.1.1.1.12.67141710 = STRING: HUAWEI
.1.3.6.1.2.1.47.1.1.1.1.12.67141774 = STRING: HUAWEI
```

there is another mib  ENTITY-MIB entAliasMappingIdentifier which maps a entPhysicalIndex to the ifIndex from IF-MIB
however this mib doesnt just return an integer with the Ifindex in it (i wish it did)

instead it returns the following:

```
.1.3.6.1.2.1.47.1.3.2.1.2.67141710.1 = OID: .1.3.6.1.2.1.2.2.1.1.35
.1.3.6.1.2.1.47.1.3.2.1.2.67141774.1 = OID: .1.3.6.1.2.1.2.2.1.1.36
.1.3.6.1.2.1.47.1.3.2.1.2.67141838.1 = OID: .1.3.6.1.2.1.2.2.1.1.37
.1.3.6.1.2.1.47.1.3.2.1.2.67141902.1 = OID: .1.3.6.1.2.1.2.2.1.1.38
.1.3.6.1.2.1.47.1.3.2.1.2.67403854.1 = OID: .1.3.6.1.2.1.2.2.1.1.4
.1.3.6.1.2.1.47.1.3.2.1.2.67436622.1 = OID: .1.3.6.1.2.1.2.2.1.1.7
.1.3.6.1.2.1.47.1.3.2.1.2.67436686.1 = OID: .1.3.6.1.2.1.2.2.1.1.8
.1.3.6.1.2.1.47.1.3.2.1.2.67436750.1 = OID: .1.3.6.1.2.1.2.2.1.1.9
```

the OID it returns is the OID for the ifIndex for each interface from IF-MIB:

```
_ifIndex OBJECT-TYPESYNTAX InterfaceIndexMAX-ACCESS read-onlySTATUS currentDESCRIPTION
```

"A unique value, greater than zero, for each interface. It is recommended that values are assigned contiguously starting from 1. The value for each interface sub-layer must remain constant at least from one re-initialization of the entity's network management system to the next re-initialization."
--_

Does anyone know if its possible to somehow cross reference the two? Ive used the generator to generate my snmp.yml for the entAliasMappingIdentifier metric and the oid for optical light levels and Prom correctly scrapes them:

```
_entAliasMappingIdentifier{auth="switch", entAliasLogicalIndexOrZero="1", entAliasMappingIdentifier="1.3.6.1.2.1.2.2.1.1.10", entPhysicalIndex="67436750", instance="xxxx", job="snmp", switch="x.x.x.x"} 1_
```
``` 
_hwEntityOpticalRxPower{auth="switch", entPhysicalIndex="67438478", instance="xxxx", job="snmp", switch="x.x.x.x} -10_
```

the end result of what im looking for is the optical light levels for each If interface. if  _entAliasMappingIdentifier_ returned just an integer for the Ifindex I could have used that and joined  the two metrics based on the _entPhysicalIndex_ but it just returns an Ifindex OID :(

I am also querying the device for the standard IF-MIB metrics so i have ifIndex etc defined as part of my snmp.yml.

thanks

Jamie

**Comment by mirackle-spb** [comment]
Did you found the answer?
